src/ObjectUtils/DwarfUtilsTest.cpp

```diff
@@ -0,0 +1,377 @@
+  [[nodiscard]] std::vector<FakeDWARFDie> children() const { return children_; }
+  [[nodiscard]] const char* getName(llvm::DINameKind /*unused*/) const { return name_; }
+  [[nodiscard]] llvm::dwarf::Tag getTag() const { return tag_; }
+  void dump() const {}
+  [[nodiscard]] FakeDWARFDie getAttributeValueAsReferencedDie(
+      llvm::dwarf::Attribute attribute) const {
+    switch (attribute) {
+      case llvm::dwarf::DW_AT_type: {
+        if (type_ != nullptr) {
+          return *type_;
+        };
+        FakeDWARFDie result{};
+        result.is_valid_ = false;
+        return result;
+      }
+      case llvm::dwarf::DW_AT_containing_type: {
+        if (containing_type_ != nullptr) {
+          return *containing_type_;
+        }
+        FakeDWARFDie result{};
+        result.is_valid_ = false;
+        return result;
+      }
+      default:
+        ORBIT_UNREACHABLE();
+    }
+  }
+
+  bool is_valid_ = true;
+  std::vector<FakeDWARFDie> children_{};
+  const char* name_{};
+  llvm::dwarf::Tag tag_ = static_cast<llvm::dwarf::Tag>(llvm::dwarf::DW_TAG_invalid);
+  const FakeDWARFDie* type_{};
+  const FakeDWARFDie* containing_type_{};
+};
+
+const FakeDWARFDie kBaseTypeDie{
+    /*is_valid=*/true,
+    /*children=*/{},
+    /*name=*/"int",
+    /*tag=*/llvm::dwarf::DW_TAG_base_type,
+    /*type=*/{},
+    /*containing_type_=*/{},
+};
+
+const FakeDWARFDie kUserSpecifiedType{
+    /*is_valid=*/true,
+    /*children=*/{},
+    /*name=*/"Foo",
+    /*tag=*/llvm::dwarf::DW_TAG_class_type,
+    /*type=*/{},
+    /*containing_type_=*/{},
+};
+
+}  // namespace
+
+namespace orbit_object_utils {
+
+TEST(DwarfTypeAsString, DiesIfInvalid) {
+  FakeDWARFDie invalid_die;
+  invalid_die.is_valid_ = false;
+
+  EXPECT_DEATH((void)DwarfTypeAsString(invalid_die), "");
+}
+
+TEST(DwarfTypeAsString, ReturnsNameOnNamedTypes) {
+  EXPECT_STREQ(DwarfTypeAsString(kBaseTypeDie).data(), "int");
+  EXPECT_STREQ(DwarfTypeAsString(kUserSpecifiedType).data(), "Foo");
+}
+
+TEST(DwarfTypeAsString, PrependsAtomicModifier) {
+  FakeDWARFDie atomic_die;
+  atomic_die.tag_ = llvm::dwarf::DW_TAG_atomic_type;
+  atomic_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(atomic_die).data(), "_Atomic int");
+}
+
+TEST(DwarfTypeAsString, PrependsConstModifier) {
+  FakeDWARFDie const_die;
+  const_die.tag_ = llvm::dwarf::DW_TAG_const_type;
+  const_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(const_die).data(), "const int");
+}
+
+TEST(DwarfTypeAsString, PrependsVolatileModifier) {
+  FakeDWARFDie volatile_die;
+  volatile_die.tag_ = llvm::dwarf::DW_TAG_volatile_type;
+  volatile_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(volatile_die).data(), "volatile int");
+}
+
+TEST(DwarfTypeAsString, PrependsRestrictModifier) {
+  FakeDWARFDie restrict_die;
+  restrict_die.tag_ = llvm::dwarf::DW_TAG_restrict_type;
+  restrict_die.type_ = &kBaseTypeDie;
+
+  EXPECT_STREQ(DwarfTypeAsString(restrict_die).data(), "restrict int");
```

Comment:
Isn't restrict usually appended? I'm not sure because it only really makes sense for pointer types. But if you had a restricted const pointer to int, I think you would write `int* const restrict`.

Reply:
If it wasn't clear; That's a Nit. Feel free to ignore.